pyccel/codegen/printing/ccode.py

```diff
@@ -1276,6 +1276,41 @@ def _print_NumpyUfuncBase(self, expr):
         code_args = ', '.join(args)
         return '{0}({1})'.format(func_name, code_args)
 
+    def _print_NumpySign(self, expr):
+        """ Print the corresponding C function for a call to Numpy.sign
+
+        Parameters
+        ----------
+            expr : Pyccel ast node
+                Python expression with Numpy.sign call
+
+        Returns
+        -------
+            string
+                Equivalent internal function in C
+
+        Example
+        -------
+            import numpy
+
+            numpy.sign(x) => isign(x)   (x is integer)
+            numpy.sign(x) => fsign(x)   (x if float)
+            numpy.sign(x) => csign(x)   (x is complex)
+
+        """
+        self.add_import(c_imports['numpy_c'])
+        func = ''
+        if isinstance(expr.dtype, NativeInteger):
+            func = 'isign'
+        elif isinstance(expr.dtype, NativeFloat):
+            func = 'fsign'
+        elif isinstance(expr.dtype, NativeComplex):
+            func = 'csign'
+        else:
+            errors.report(f'{PYCCEL_RESTRICTION_TODO} ({expr.dtype}[kind = {expr.precision}])', severity='fatal')
```

Comment:
Please add the `symbol` keyword to this error call (this will allow pyccel to determine the line where the error is coming from). As you are writing your own error message rather than relying on the auto-generated message raised when calling `NotImplementedError` I think it would be better to provide more information in the error. For example it is not obvious from this error that it is coming from the `sign` function

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/7e6dc6a384ba3dc2a0b2cca80ae60993d79eee08 checking for dtype inside the class now https://github.com/pyccel/pyccel/pull/1169/commits/0d6ac52998719e9972a6177a0033ca6413b8030a